pagopa/io-app
# chore(Sicilia Vola): [IASV-46] Download voucher pdf

## THIS PR DEPENDS ON #3429 
## Short description
This PR adds the management of saving the voucher in pdf format.
This is used both in the `VoucherDetailsScreen` and in `VoucherGeneratedScreen`.

## List of changes proposed in this pull request
- Added the client and the handler to manage the saving of the voucher in pdf format
- Added permission in order to store the file in the document directory (more info [here](https://stackoverflow.com/questions/54626359/react-native-fs-ios-not-showing-files-in-the-document-directory) )
- Added the `svGetPdfVoucher.request` action to both the `VoucherDetailsScreen` and in `VoucherGeneratedScreen` screens.


https://user-images.githubusercontent.com/11773070/137708428-f0ba234c-b71f-4b45-be55-59706d833329.mp4


## How to test
Starts the flow from the SiciliaVola service detail press the Voucher List button.
Enter in the voucher details, press the `Open QR code` button the press the `Save` button.

Make sure to use this branch on the dev server [IASV-46-sv-save-voucher](https://github.com/pagopa/io-dev-api-server/tree/IASV-46-sv-save-voucher)
